homeassistant/components/lgthinq/__init__.py

```diff
@@ -0,0 +1,331 @@
+"""Support for LG ThinQ Connect device."""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Collection
+from dataclasses import dataclass, field
+import logging
+from typing import Any
+
+from thinqconnect.thinq_api import ThinQApiResponse
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_DEVICE_ID,
+    CONF_ACCESS_TOKEN,
+    CONF_COUNTRY,
+    Platform,
+)
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+    callback,
+)
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryError
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util.json import JsonValueType
+
+from .const import (
+    CONF_CONNECT_CLIENT_ID,
+    DEFAULT_COUNTRY,
+    DOMAIN,
+    SERVICE_ATTR_DEVICE_INFO,
+    SERVICE_ATTR_RESULT,
+    SERVICE_ATTR_VALUE,
+)
+from .device import LGDevice, async_setup_lg_device
+from .thinq import ThinQ
+
+type ThinqConfigEntry = ConfigEntry[ThinqData]
+
+
+@dataclass(kw_only=True)
+class ThinqData:
+    """A class that holds runtime data."""
+
+    device_map: dict[str, LGDevice] = field(default_factory=dict)
+
+
+PLATFORMS = [
+    Platform.SWITCH,
+]
+
+SERVICE_GET_DEVICE_PROFILE_SCHEMA = vol.Schema(
+    {vol.Required(ATTR_DEVICE_ID): cv.string}
+)
+SERVICE_GET_DEVICE_STATUS_SCHEMA = vol.Schema({vol.Required(ATTR_DEVICE_ID): cv.string})
+SERVICE_POST_DEVICE_STATUS_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_DEVICE_ID): cv.string,
+        vol.Required(SERVICE_ATTR_VALUE): vol.Any(cv.string, dict),
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ThinqConfigEntry) -> bool:
+    """Set up an entry."""
+    _LOGGER.warning("Starting set up entry")
+
+    # Validate entry data.
+    client_id = entry.data.get(CONF_CONNECT_CLIENT_ID)
+    if not isinstance(client_id, str):
+        raise ConfigEntryError(f"Invalid client id: {client_id}")
+
+    access_token = entry.data.get(CONF_ACCESS_TOKEN)
+    if not isinstance(access_token, str):
+        raise ConfigEntryAuthFailed(f"Invalid PAT: {access_token}")
+
+    # Initialize runtime data.
+    entry.runtime_data = ThinqData()
+
+    thinq = ThinQ(
+        client_session=async_get_clientsession(hass),
+        country_code=entry.data.get(CONF_COUNTRY, DEFAULT_COUNTRY),
+        client_id=client_id,
+        access_token=access_token,
+    )
+    device_registry = dr.async_get(hass)
```

Comment:
Not fully understanding why the device registry is retrieved in this location. Probably better to move this into the places it is actually used (since you pass along `hass` anyways, you can do it locally).

Reply:
We thought we better call it once than twice.

Reply:
Moved into the each method.